ztools/ztls/handshake_messages.go

```diff
@@ -99,6 +101,17 @@ func (m *clientHelloMsg) marshal() []byte {
 		extensionsLength += 1
 		numExtensions++
 	}
+	if len(m.alpnProtocols) > 0 {
+		extensionsLength += 2
+		for _, s := range m.alpnProtocols {
+			if l := len(s); l == 0 || l > 255 {
+				panic("invalid ALPN protocol")
```

Comment:
If you think panic() is not appropriate, then it would be easy to remove that check entirely, and I can modify the PR for it.
That will result in client sending invalid ALPN to server, which might or might not fail the handshake.